extensions/wikia/TemplateClassification/scripts/TemplateClassificationInCategory.js

```diff
@@ -0,0 +1,83 @@
+/**
+ * TemplateClassificationInView module
+ *
+ * Works for view page.
+ *
+ * Provides selected type for TemplateClassificationModal and handles type submit
+ */
+define('TemplateClassificationInCategory', ['jquery', 'mw', 'wikia.nirvana', 'wikia.tracker', 'TemplateClassificationModal', 'BannerNotification'],
+	function ($, mw, nirvana, tracker, templateClassificationModal, BannerNotification) {
+		'use strict';
+
+		var $typeLabel;
+
+		function init() {
+
+			$typeLabel = $('.template-classification-type-text');
+			templateClassificationModal.init(getType, sendClassifyTemplatesRequest, 'bulkEditType');
+
+			$typeLabel.on('mousedown', function () {
+				tracker.track({
+					trackingMethod: 'analytics',
+					category: 'template-classification-entry-point',
+					action: tracker.ACTIONS.CLICK,
+					label: 'category-view-page'
+				});
+			});
+		}
+
+		function getType() {
+			return '';
+		}
+
+		function sendClassifyTemplatesRequest(selectedTemplateType) {
+			nirvana.sendRequest({
+				controller: 'TemplateClassification',
+				method: 'classifyTemplateByCategory',
+				data: {
+					type: selectedTemplateType,
+					category: mw.config.get('wgTitle'),
+					editToken: mw.user.tokens.get('editToken')
+				},
+				callback: function () {
+					var notification = new BannerNotification(
+						mw.message('template-classification-edit-modal-success').escaped(),
+						'confirm'
+					);
+
+					notification.show();
+				},
+				onErrorCallback: function (error) {
+					var errorMessage, e;
+
+					if (error.responseText.length) {
+						e = $.parseJSON(error.responseText);
+						errorMessage = e.exception.message;
+					} else {
+						errorMessage = mw.message('template-classification-edit-modal-error').escaped();
+					}
+
+					var notification = new BannerNotification(errorMessage, 'error');
```

Comment:
All variables should be declared at the top of the scope context per conventions.